### PR TITLE
Make sure we return the classes on a fail.

### PR DESCRIPTION
**Pull request by Camwyn**
[BTRIA-416]
📷 https://dl.dropbox.com/s/c41ok1fucn9ighh/shot_200630_235308.jpg 😢 
📸 http://p.tri.be/pnSol2 😃 

[BTRIA-416]: https://moderntribe.atlassian.net/browse/BTRIA-416

### CODE REVIEW

File: src/Tribe/Utils/Body_Classes.php [hunk 1]
```diff
@@ -287,15 +287,15 @@ public function add_body_classes( $classes = [] ) {
 	 * @return array|false Current list of admin body classes if added, otherwise false.
```
Comment (paulmskim) [inline src/Tribe/Utils/Body_Classes.php:287]: should this be `@return array<string>`?

File: src/Tribe/Utils/Body_Classes.php [hunk 2]
```diff
@@ -287,15 +287,15 @@ public function add_body_classes( $classes = [] ) {
...
-		if ( ! $this->should_add_body_classes( $this->get_class_names( 'admin' ), (array) $classes, 'admin' ) ) {
-			return false;
+		if ( ! $this->should_add_body_classes( $this->get_class_names( 'admin' ), (array) $existing_classes, 'admin' ) ) {
+			return $classes;
 		}
 
 		$element_classes = new Element_Classes( $this->get_class_names( 'admin' ) );
 
-		return implode( ' ', array_merge( $classes, $element_classes->get_classes() ) );
+		return implode( ' ', array_merge( $existing_classes, $element_classes->get_classes() ) );
```
Comment (paulmskim) [inline src/Tribe/Utils/Body_Classes.php:298]: Are we returning an array? or a string?

**Comment by paulmskim** [comment]
@Camwyn do we have tests for this? If not, could we add some?

File: src/Tribe/Utils/Body_Classes.php (hunk 1 above)
Comment (Camwyn) [reply src/Tribe/Utils/Body_Classes.php:287]: yep

**Comment by Camwyn** [comment]
@paulmskim There are some tests - unfortunately I didn't add any specific to the admin classes. Going to do that here as well.
https://github.com/moderntribe/tribe-common/blob/master/tests/wpunit/Tribe/Body_ClassesTest.php

File: src/Tribe/Service_Providers/Body_Classes.php [hunk 3]
```diff
@@ -20,6 +20,7 @@ class Body_Classes extends \tad_DI52_ServiceProvider {
 	 */
 	public function register() {
 		tribe_singleton( Body_Class_Object::class, Body_Class_Object::class );
+		tribe_singleton( 'tribe_body_classes_provider', $this );
```
Comment (bordoni) [inline src/Tribe/Service_Providers/Body_Classes.php:23]: Check the naming standards for common: https://moderntribe.github.io/products-engineering/docs/plugins/namespacing/

File: src/Tribe/Service_Providers/Body_Classes.php [hunk 4]
```diff
@@ -20,6 +20,7 @@ class Body_Classes extends \tad_DI52_ServiceProvider {
 	 */
 	public function register() {
 		tribe_singleton( Body_Class_Object::class, Body_Class_Object::class );
+		tribe_singleton( 'common.body_classes_provider', $this );
```
Comment (bordoni) [inline src/Tribe/Service_Providers/Body_Classes.php:23]:
```suggestion
		tribe_singleton( 'common.service_providers.body_classes', $this );
```

**Comment by Camwyn** [comment]
@sc0ttkclark why the question tag?

**Comment by Camwyn** [comment]
@bordoni I keep having to update this when I get back to it - can I get a re-approval please?

**Pull request merged**